src/main/java/com/proximyst/ban/utils/ArrayUtils.java

```diff
@@ -0,0 +1,35 @@
+//
+// ban - A punishment suite for Velocity.
+// Copyright (C) 2020 Mariell Hoversholm
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+package com.proximyst.ban.utils;
+
+import java.util.Arrays;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+public final class ArrayUtils {
+  private ArrayUtils() throws IllegalAccessException {
+    throw new IllegalAccessException(getClass().getSimpleName() + " cannot be instantiated.");
+  }
+
+  @SafeVarargs
+  public static <T> T @NonNull [] append(final T @NonNull [] array, final T @NonNull ... items) {
```

Comment:
Uh-oh, stinky. Not only is appending to an array kinda disgusting (but I'm guessing this is your only option), this can also return some funky results. This code will still run if `array` and `items` are of type `A` and `B`, respectively, when `A` and `B` extend from the same base class (for example, Double and Integer both extend from the `Number` class). This will result in an exception.

Add a type parameter, please :)

Or, use Guava's `ObjectArrays#concat`/`<PrimitiveName>s#concat`

Reply:
If `T` is `Double`, you can't add e.g. `Integer`; that'd require a `Number` instead, in which case you get exactly that. The types are safe.

Reply:
That is what you'd expect. However, T is resolved to `? extends BaseClass`, so `? extends Number` in this case. Running the following will result in an ArrayStoreException:
```
    Double[] x = {0.0, 1.1};
    Integer[] y = {1, 2};

    System.out.println(Arrays.toString(ArrayUtils.append(x, y)));
```

Reply:
After actually using my brain, I concluded that ArrayUtils.append will take *any* combination of non-primitive types, as long as they're both arrays, because they extend Object.